peers/app_request_network.go

```diff
@@ -273,6 +277,7 @@ func NewNetwork(
 	}
 	vdrsCache := cache.NewTTLCache[ids.ID, avalancheWarp.CanonicalValidatorSet](canonicalValidatorSetCacheTTL)
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
We might also want to consider having `NewNetwork` take a `ctx` parameter that we can use here instead of `context.Background()`